runtime/autoload/provider/clipboard.vim

```diff
@@ -4,43 +4,88 @@
 let s:copy = {}
 let s:paste = {}
 
+" when caching is enabled, store the jobid of the xclip/xsel process
+" keeping ownership of the selection, so we know how long the cache is valid
+let s:selection = { 'owner': 0, 'data': [] }
+
+function! s:selection.on_exit(jobid, data, event)
+  " At this point this nvim instance might already have launched
+  " a new provider instance. Don't drop ownership in this case.
+  if self.owner == a:jobid
+    let self.owner = 0
+  endif
+endfunction
+
+let s:selections = { '*': s:selection, '+': copy(s:selection)}
+
 function! s:try_cmd(cmd, ...)
-  let out = a:0 ? systemlist(a:cmd, a:1, 1) : systemlist(a:cmd, [''], 1)
+  let argv = split(a:cmd, " ")
+  let out = a:0 ? systemlist(argv, a:1, 1) : systemlist(argv, [''], 1)
   if v:shell_error
-    echo "clipboard: error: ".(len(out) ? out[0] : '')
-    return ''
+    echohl WarningMsg
+      echo "clipboard: error: ".(len(out) ? out[0] : '')
+    echohl None
+    return 0
   endif
   return out
 endfunction
 
+let s:cache_enabled = 1
```

Comment:
Perhaps, but is serves as a default for the value in the executable detection just below. If it's clearer, I can add `let s:cache_enabled = 1` to the X11 branches instead.